wire-ios-system/Tests/SplitViewControllerDelegate/SupportedOrientationsDelegatingSplitViewControllerDelegateTests.swift

```diff
@@ -0,0 +1,146 @@
+    func testAllOrientationsSupported() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .doubleColumn)
+        splitViewController.setViewController(ViewController(), for: .primary)
+        splitViewController.setViewController(ViewController(), for: .secondary)
+
+        // When
+        let result = sut.splitViewControllerSupportedInterfaceOrientations(splitViewController)
+
+        // Then
+        XCTAssertEqual(result, .all)
+    }
+
+    @MainActor
+    func testOnlyUpsideDownSupported() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .tripleColumn)
+        splitViewController.setViewController(ViewController([.portrait, .portraitUpsideDown]), for: .primary)
+        splitViewController.setViewController(ViewController([.landscapeLeft, .portraitUpsideDown]), for: .supplementary)
+        splitViewController.setViewController(ViewController([.landscapeRight, .portraitUpsideDown]), for: .secondary)
+
+        // When
+        let result = sut.splitViewControllerSupportedInterfaceOrientations(splitViewController)
+
+        // Then
+        XCTAssertEqual(result, .portraitUpsideDown)
+    }
+
+    @MainActor
+    func testAllButUpsideDownSupported() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .doubleColumn)
+        splitViewController.setViewController(ViewController([.portrait, .portraitUpsideDown]), for: .primary)
+        splitViewController.setViewController(ViewController([.landscapeRight, .portraitUpsideDown]), for: .secondary)
+
+        // When
+        let result = sut.splitViewControllerSupportedInterfaceOrientations(splitViewController)
+
+        // Then
+        XCTAssertEqual(result, .portraitUpsideDown)
+    }
+
+    @MainActor
+    func testAllSupportedWhenNoViewControllers() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .tripleColumn)
+
+        // When
+        let result = sut.splitViewControllerSupportedInterfaceOrientations(splitViewController)
+
+        // Then
+        XCTAssertEqual(result, .all)
+    }
+
+    @MainActor
+    func testDelegateIsSet() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .tripleColumn)
+
+        // When
+        sut.setAsDelegateAndNontomicRetainedAssociatedObject(splitViewController)
+
+        // Then
+        XCTAssert(splitViewController.delegate === sut)
+    }
+
+    @MainActor
+    func testDelegateIsRetained() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .tripleColumn)
+
+        // When
+        sut.setAsDelegateAndNontomicRetainedAssociatedObject(splitViewController)
+        weak var weakSut = sut
+        sut = nil
+
+        // Then
+        withExtendedLifetime(splitViewController) {
+            XCTAssertNotNil(weakSut)
+        }
+    }
+}
+
+// MARK: - ViewController
+
+private final class ViewController: UIViewController {
```

Comment:
Optional: Consider making this an internal test helpers `ViewController` and reusing in `SupportedOrientationsDelegatingNavigationControllerDelegateTests`

Reply:
I think we discussed about this in the past and we would like to have unit test files to have little dependencies, so that you could theoretically copy them into different modules and they would still work.